### PR TITLE
Ctpps pixel clusterizer

### PR DESCRIPTION
**Pull request by fabferro**
CTPPS pixel clusterizer with calibration done reading data from DB.
Also cleaning corresponding CondFormats from previous PR.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by slava77** [comment]
@fabferro 
will the tests still fail without the DB payloads in the GT available in IB? If so, when do we expect it to appear?

About the name RecoCTPPS/CTPPSPixelLocal: do we expect any other "Pixel" in the CT-PPS system? RecoCTPPS/PixelLocal would be a bit less redundantly named and easier to read.

**Comment by fabferro** [comment]
@slava77 : as far as I understood from @arunhep mail things are going to be readied ~now. That's why I re-submitted the PR's.

As to the renaming CTPPSPixelLocal->PixelLocal, I'll do it.

**Comment by arunhep** [comment]
@fabferro we need same tags for MC as well, as mentioned by @slava77 in email.
Because the same code runs for 2016 MC as well as Phase-1 workflows.
So, please prepare the tags with IOV set to 1 and name them _mc instead of _hlt.

File: RecoCTPPS/CTPPSPixelLocal/src/RPixDetClusterizer.cc [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+	}
+
+      }
+    }
+	     
+  }  // while accretion
+
+
+ endClus:
```
Comment (slava77) [inline RecoCTPPS/CTPPSPixelLocal/src/RPixDetClusterizer.cc]: what is this?

Comment (fabferro) [inline RecoCTPPS/CTPPSPixelLocal/src/RPixDetClusterizer.cc]: The same as this: https://github.com/cms-sw/cmssw/blob/master/RecoLocalTracker/SiPixelClusterizer/plugins/PixelThresholdClusterizer.cc#L459
I don't like it much but I took it as it is.

**Comment by fabferro** [comment]
CTPPSPixelLocal->PixelLocal done

File: RecoCTPPS/PixelLocal/plugins/CTPPSPixelClusterProducer.cc [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  edm::Handle<edm::DetSetVector<CTPPSPixelDigi> > rpd;
+  iEvent.getByToken(tokenCTPPSPixelDigi_, rpd);
+
+// get analysis mask to mask channels
+  edm::ESHandle<CTPPSPixelAnalysisMask> aMask;
+  iSetup.get<CTPPSPixelAnalysisMaskRcd>().get("RPix",aMask);
+  
+// get calibration DB
+
+ theGainCalibrationDB.getDB(iEvent,iSetup);
```
Comment (slava77) [inline RecoCTPPS/PixelLocal/plugins/CTPPSPixelClusterProducer.cc]: as mentioned in email, the DB reads can be protected behind ```if (rpd->size())```

Comment (fabferro) [inline RecoCTPPS/PixelLocal/plugins/CTPPSPixelClusterProducer.cc]: added the protection for calibration and mask. Please check. thanks

**Comment by slava77** [comment]
@cmsbuild please test

jenkins jobs should complete now (no CTPPS pixel in inputs used for tests .. we should get an update to workflow set soon though)

**Comment by arunhep** [comment]
@fabferro please cherry-pick this commit https://github.com/arunhep/cmssw/commit/2ae8cbba3becd8d17529180527ad333a6c5c79a5
it contains the GT updates for CTPPS Pixel tags for data GTs.
we then can re-trigger the tests.

there is another version of autoCond open for #19307 and whichever PR will be merged earlier, the other one will need a rebase

**Comment by slava77** [comment]
-1
needs a proper rebase
http://cms-sw.github.io/tutorial-resolve-conflicts.html

it also looks like after addition of https://github.com/cms-sw/cmssw/pull/19460/commits/9f247881aa0ecb508757c95074b2ff650e8eedd1 via a merge the topic branch for this PR can not work anymore for both 92X and 93x/master.

I'd suggest to go back to the last state of this PR before the rebase attempt and cherry=pick the autoCond commit

**Comment by slava77** [comment]
here, on top of the master, there are still two merges that do not belong: 
``` 
 	Merge pull request #19417
 	Merge pull request #19452
```
it looks like there was a merge of the master and then rebase to an IB older than these two commits in the master.
So, this is not so good either (even though it's not completely wrong).

**Comment by fabferro** [comment]
Yes, I see. I start to be lost about how to cure it...

**Comment by ianna** [comment]
-1

proper rebase should remove the need for geometry signature

**Comment by fabferro** [comment]
Closing PR. Starting brand new PR #19470 for 93x. Adding also @arunhep 's changes to autoCond.py

**Pull request closed without merging**